src/zeta/aca_oam_server.cpp

```diff
@@ -147,57 +178,59 @@ string ACA_Oam_Server::get_tunnel_id(uint8_t *vni){
   return vpc_id;
 }
 
-string ACA_Oam_Server::get_vpc_id(uint8_t *vni){
-  string vpc_id;
-  stringstream ss;
+oam_match ACA_Oam_Server::get_oam_match_field(oam_message *oammsg){
+  oam_match match;
 
-  for(int i = 0; i < 6; i++){
-    ss << std::hex << std::setw(2) << std::setfill('0')
-         << static_cast<unsigned int>(vni[i]);
-  }
+  flow_inject_msg msg_data = oammsg->data.msg_inject_flow;
 
-  ss >> vpc_id;
-  vpc_id.pop_back();
+  match.sip = inet_ntoa(msg_data.inner_src_ip);
+  match.dip = inet_ntoa(msg_data.inner_dst_ip);
+  match.sport = to_string(ntohs(msg_data.src_port));
+  match.dport = to_string(ntohs(msg_data.dst_port));
+  match.proto = to_string(msg_data.proto);
+  match.vni = get_vpc_id(msg_data.vni);
 
-  return vpc_id;
+  return match;
+}
+
+oam_action ACA_Oam_Server::get_oam_action_field(oam_message *oammsg){
+  oam_action action;
+
+  flow_inject_msg msg_data = oammsg->data.msg_inject_flow;
+
+  action.inst_nw_dst = inet_ntoa(msg_data.inst_dst_ip);
+  action.node_nw_dst = inet_ntoa(msg_data.node_dst_ip);
+  action.inst_dl_dst = _get_mac_addr(msg_data.inst_dst_mac);
+  _standardize_mac_address(action.inst_dl_dst);
+  action.node_dl_dst = _get_mac_addr(msg_data.node_dst_mac);
+  _standardize_mac_address(action.node_dl_dst);
+  action.idle_timeout = to_string(msg_data.idle_timeout);
+
+  return action;
 }
 
-void ACA_Oam_Server::_parse_oam_flow_injection(uint32_t in_port, oam_message *oammsg)
+void ACA_Oam_Server::_parse_oam_flow_injection(oam_message *oammsg)
 {
   unsigned long not_care_culminative_time;
   int overall_rc = EXIT_SUCCESS;
 
-  flow_inject_msg opdata = oammsg->data.msg_inject_flow;
-  string remote_host_ip = inet_ntoa(opdata.node_dst_ip);
-  string vpc_id = get_tunnel_id(opdata.vni);
-  string tunnel_id = get_tunnel_id(opdata.vni);
+  oam_match match = get_oam_match_field(oammsg);
+  oam_action action = get_oam_action_field(oammsg);
 
-  string inst_dst_mac = _standardize_mac_address(const_cast<string*>(ether_ntoa(opdata.inst_dst_mac)));
+  string remote_host_ip = action.node_nw_dst;
+  string vpc_id = match.vni;
 
   alcor::schema::NetworkType network_type = alcor::schema::NetworkType::VXLAN;
 
   if(!aca_is_port_on_same_host(remote_host_ip)){
-      ACA_LOG_INFO("port_neighbor not exist!\n");
+      ACA_LOG_INFO("%s", "port_neighbor not exist!\n");
       aca_ovs_l2_programmer::ACA_OVS_L2_Programmer::get_instance().create_update_neighbor_port(vpc_id,
```

Comment:
I am not sure if we can reuse this function create_update_neighbor_port because the previous neighbor update logic is different from Zeta design. 

The previous create_update_neighbor_port logic will enable **all traffic** from local port to neighbor VM.

Zeta logic only install direct **communcation flows** only based on the matcher fields inside OAM packet.